tsig.go

```diff
@@ -98,14 +148,13 @@ type timerWireFmt struct {
 // timersOnly is false.
 // If something goes wrong an error is returned, otherwise it is nil.
 func TsigGenerate(m *Msg, secret, requestMAC string, timersOnly bool) ([]byte, string, error) {
+	return tsigGenerateProvider(m, requestMAC, timersOnly, tsigHMACProvider(secret))
+}
+
+func tsigGenerateProvider(m *Msg, requestMAC string, timersOnly bool, provider TsigProvider) ([]byte, string, error) {
```

Comment:
I think switch the order of parameters here, move `provider TsigProvider` between `m *Msg` and `requestMAC string`. That way the secret-related part is in the same place as it is for `TsigGenerate`.

Reply:
Done. Originally I kept the order so it was `TsigGenerate()` plus a provider but with the original secret parameter changed this makes more sense.